src/functions/L1plusL2.jl

```diff
@@ -0,0 +1,56 @@
+# (weighted) sum of L2 norm and L1 norm
+# for Group Lasso, this can be used together with src/calculus/slicedSeparableSum
+
+using ProximalOperators
+import ProximalOperators: prox!
+
+export L1plusL2
+
+"""
+**L1 norm plus L2 norm**
+
+    L1plusL2(λ_1=1, λ_2=1)
+
+With two nonnegative scalar parameters λ_1 and λ_2, returns the function
+```math
+f(x) = λ_1 ∑_{i=1}^{n} |x_i| + λ_2 \\sqrt{x_1^2 + … + x_n^2}.
+```
+"""
+struct L1plusL2{L <: Real, M <: Real} <: ProximableFunction
+    lambda1::L
+    lambda2::M
+    function L1plusL2{L, M}(lambda1::L, lambda2::M) where {L <: Real, M <: Real}
+        if lambda1 < 0 || lambda2 < 0
+            error("parameters λ_1, λ_2 must be nonnegative")
+        else
+            new(lambda1, lambda2)
+        end
+    end
+end
+
+is_separable(f::L1plusL2) = false
+is_convex(f::L1plusL2) = true
+is_positively_homogeneous(f::L1plusL2) = true
+
+L1plusL2(lambda1::L=1, lambda2::M=1) where {L <: Real, M <: Real} = L1plusL2{L, M}(lambda1, lambda2)
+
+function (f::L1plusL2)(x::AbstractArray)
+    return f.lambda1 * norm(x, 1) + f.lambda2 * norm(x, 2)
+end
+
+function prox!(y::AbstractArray{T}, f::L1plusL2, x::AbstractArray{T}, gamma::Real=1) where T <: Real
+
+    f1 = NormL1(f.lambda1)
+    f2 = NormL2(f.lambda2)
+
+    y0, t = prox_naive(f1, x, gamma)
```

Comment:
`prox_naive` is generally intended to be a simple implementation used to test the functionality, but it is also often inefficient, so we don't want to use it in our `prox!` implementations.

The simple way to solve this is to do
```julia
y0 = similar(y)
prox!(y0, f1, x, gamma)
```

However, the implementation on prox! on NormL2 allows aliasing, so even better is simply
```julia
prox!(y, f1, x, gamma)
prox!(y, f2, y, gamma)
```

Reply:
Thanks! I was not sure whether having `y` as input and writing the result  into `y` could potentially cause problems. But then I will change it.

Reply:
You are right that it can be dangerous in general, but in this case it will only be called on NormL2, and the implementation here looks safe in that regard:
https://github.com/JuliaFirstOrder/ProximalOperators.jl/blob/c9a250f8836e41450bada594ed23baa22dc8e928/src/functions/normL2.jl#L39